YOZM/YOZM/Service/CloudKit/CloudKitService.swift

```diff
@@ -0,0 +1,85 @@
+//
+//  CloudKitService.swift
+//  YOZM
+//
+//  Created by 정희균 on 8/15/25.
+//
+
+import CloudKit
+
+/// CloudKit 서비스 에러
+enum CloudKitError: Error, LocalizedError {
+    case recordNotFound
+    case audioDataNotFound
+    case audioURLNotFound
+    case missingField(field: String)
+    case invalidData(String)
+    
+    var errorDescription: String {
+        switch self {
+        case .recordNotFound:
+            return "레코드를 찾을 수 없습니다"
+        case .audioDataNotFound:
+            return "음성 데이터를 찾을 수 없습니다"
+        case .audioURLNotFound:
+            return "음성 파일 URL을 찾을 수 없습니다"
+        case .missingField(field: let field):
+            return "필드 \(field)가 누락되었습니다"
+        case .invalidData(let message):
+            return "잘못된 데이터: \(message)"
+        }
+    }
+}
+
+final class CloudKitService {
+    private let uploader = CloudKitUploaderService()
+    private let fetcher = CloudKitFetcherService()
+    private let audio = CloudKitAudioService()
```

Comment:
싱글톤으로 바꾸는 것이 어떨까요?

Reply:
이미 CloudKitService가 싱글톤이라서 `CloudKitUploaderService, CloudKitFetcherService, CloudKitAudioService`까지 싱글톤으로 처리하면 싱글톤이 중첩되는 것 같아요!

CloudKitService가 싱글톤이라서 내부의 uploader, fetcher, audio 인스턴스들도 앱 전체에서 하나씩만 존재하는 상태인데요.
그래서 메모리 효율성은 이미 달성한 것 같은데 혹시 싱글톤으로 처리하고 싶은 이유가 있으실까요??

Reply:
각 서비스가 `CloudKitService`를 위한 서비스이고, 하나만 존재할 거라면 다른 곳에서 인스턴스를 잘못 생성할 수 없게 싱글톤으로 만드는 것이 좋다고 생각했는데, 그렇게 할 필요는 없을 것 같습니다! 오히려 `extension`으로 `CloudKitService`에 각 서비스를 안에 넣는 방법도 있을 것 같아요.

Reply:
extension으로 각 서비스 처리했습니다!!
```suggestion
    final class CloudKitService {
    private let uploader = CloudKitUploaderService()
    private let fetcher = CloudKitFetcherService()
    private let audio = CloudKitAudioService()
    
    static let shared = CloudKitService()
    private init() {}
}

// MARK: - CloudKitUploaderService Management
extension CloudKitService {
    func upload() async throws {
        let chapters = try loadChaptersFromJSON()
        try await uploader.upload(chapters: chapters)
    }
    
    private func loadChaptersFromJSON() throws -> [Chapter] {
        guard let url = Bundle.main.url(forResource: "exampleData", withExtension: "json") else {
            throw CloudKitError.invalidData("exampleData.json 파일을 찾을 수 없습니다")
        }
        
        let data = try Data(contentsOf: url)
        let chapterData = try JSONDecoder().decode(ChapterData.self, from: data)
        
        return chapterData.chapters
    }
}

// MARK: - CloudKitFetcherService Management
extension CloudKitService {
    func fetchChapter(by id: Int64) async throws -> Chapter {
        return try await fetcher.fetchChapter(by: id)
    }
}
```